helm/cluster-api-provider-azure/templates/deployment.yaml

```diff
@@ -0,0 +1,93 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    aadpodidbinding: capz-controller-aadpodidentity-selector
+    control-plane: capz-controller-manager
+    {{- include "labels.common" . | nindent 4 }}
+  name: capz-controller-manager
+  namespace: capz-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      control-plane: capz-controller-manager
```

Comment:
nit: `control-plane: capz-controller-manager` is used in both selectors, how about moving it to `"labels.selector"` in `_helpers.tpl`?

Reply:
I was thinking about that, but then `labels.selector` is included in `labels.common`, which is then included in many more places, so that would effectively add `control-plane: capz-controller-manager` all over the place.

I asked in CAPZ channel about this label, so we understand a bit more why/where it is needed https://kubernetes.slack.com/archives/CEX9HENG7/p1614854690112900.